dsp/modules/aws_providers.py

```diff
@@ -1,17 +1,20 @@
 """AWS providers for LMs."""
 
+import importlib.util
 from abc import ABC, abstractmethod
 from typing import Any, Optional
 
 import backoff
+
 from dsp.utils.settings import settings
 
-try:
-    import boto3
+boto3_spec = importlib.util.find_spec("boto3")
```

Comment:
I think the previous try-except logic suffices? fine to merge after reverting back this logic

Reply:
Hello @arnavsinghvi11. I reversed the logic as you suggested but it was necessary to remove the boto3 import here because ruff was failing. We are already importing boto3 in the AWSProvider class constructor.